pkg/cache/sync/sync.go

```diff
@@ -65,6 +68,15 @@ func NewSyncManager(handler SyncHandler, coreClient client.WithWatch) *SyncManag
 
 // Start begins the sync process with periodic and one-time scheduling
 func (m *SyncManager) Start(ctx context.Context, defaultSyncFrequency time.Duration) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	if m.running {
+		klog.Warningf("repositorySync %+v: already running, ignoring start request", m.handler.Key())
```

Comment:
This is dangerous, as updates in quick succession to the Repository resource will be lost. Would be better to filter irrelevant change events if something can start multiple sync threads.